model/AbstractRdsResultStorage.php

```diff
@@ -616,7 +632,33 @@ protected function unserializeVariableValue($value)
      */
     protected function serializeVariableValue($value)
     {
-        return serialize($value);
+        $serializedValue = json_decode(json_encode($value), true);
```

Comment:
Why do we need to encode and decode again?

Reply:
Can we test if `$value` implements JsonSerializable instead of encoding, decoding, encoding?
Variable serialization is part of the critical path (happens while the test is running, and every but of performance matters)

Reply:
It was necessary to do like this to determine the type. After moving the type check to the variable class, this is not needed anymore here. I fixed it.